controllers/keda/scaledobject_controller.go

```diff
@@ -132,7 +132,14 @@ func (r *ScaledObjectReconciler) SetupWithManager(mgr ctrl.Manager, options cont
 				predicate.GenerationChangedPredicate{},
 			),
 		)).
-		Owns(&autoscalingv2.HorizontalPodAutoscaler{}).
+		// Trigger a reconcile only when the HPA spec,label or annotation changes. 
+		// Ignore updates to HPA status
+		Owns(&autoscalingv2.HorizontalPodAutoscaler{}, builder.WithPredicates(
+			predicate.Or(
+				predicate.LabelChangedPredicate{},
+				predicate.AnnotationChangedPredicate{},
+				kedacontrollerutil.HPAUpdatePredicate{},
```

Comment:
```suggestion
				predicate.GenerationChangedPredicate{},
```
I wonder, isn't this enough?

Reply:
I checked for this. Actually when we do kubectl get hpa <name> -oyaml, there is no generation field in the metadata. So, this might not work.

Reply:
That is why i had to add a custom predicate.

Reply:
I mean having this existing predicate instead of the one you added `kedacontrollerutil.HPAUpdatePredicate{},` together with the one for labels and annotations.

Reply:
wow, interesting 🤔 I wonder why it doesn't have the field.

Reply:
should we increment the generation from our side?